lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -6,6 +6,40 @@ class JsonApi < Base
         autoload :PaginationLinks
         autoload :FragmentCache
 
+        # TODO: if we like this abstraction and other API objects to it,
+        # then extract to its own file and require it.
+        module ApiObjects
+          module JsonApi
+            ActiveModel::Serializer.config.jsonapi_version = '1.0'
+            ActiveModel::Serializer.config.jsonapi_toplevel_meta = {}
+            # Make JSON API top-level jsonapi member opt-in
+            # ref: http://jsonapi.org/format/#document-top-level
+            ActiveModel::Serializer.config.jsonapi_include_toplevel_object = false
+            module_function
+
+            def add!(hash)
+              hash.merge!(object) if include_object?
+            end
+
+            def include_object?
+              ActiveModel::Serializer.config.jsonapi_include_toplevel_object
+            end
+
+            # TODO: see if we can cache this
+            def object
```

Comment:
since we add meta in some of the tests, I'd have to write a way to clear the cache, which I don't want to do right now, could be in a test helper I guess.